File: admob to GAM/java/adapter/AdNetworkCustomEvent.java


import android.content.Context;

import androidx.annotation.NonNull;

import com.example.netlinkiron.admob.BannerCustomEventLoader;
import com.example.netlinkiron.admob.InterstialCustomEvent;
import com.example.netlinkiron.admob.NativeCustomEvent;
import com.example.netlinkiron.admob.RewardCustomEvent;
import com.google.android.gms.ads.mediation.Adapter;
import com.google.android.gms.ads.mediation.InitializationCompleteCallback;
import com.google.android.gms.ads.mediation.MediationAdLoadCallback;
import com.google.android.gms.ads.mediation.MediationBannerAd;
import com.google.android.gms.ads.mediation.MediationBannerAdCallback;
import com.google.android.gms.ads.mediation.MediationBannerAdConfiguration;
import com.google.android.gms.ads.mediation.MediationConfiguration;
import com.google.android.gms.ads.mediation.MediationInterstitialAd;
import com.google.android.gms.ads.mediation.MediationInterstitialAdCallback;
import com.google.android.gms.ads.mediation.MediationInterstitialAdConfiguration;
import com.google.android.gms.ads.mediation.MediationNativeAdCallback;
import com.google.android.gms.ads.mediation.MediationNativeAdConfiguration;
import com.google.android.gms.ads.mediation.MediationRewardedAd;
import com.google.android.gms.ads.mediation.MediationRewardedAdCallback;
import com.google.android.gms.ads.mediation.MediationRewardedAdConfiguration;
import com.google.android.gms.ads.mediation.UnifiedNativeAdMapper;
import com.google.android.gms.ads.mediation.VersionInfo;

import java.util.List;

public class AdNetworkCustomEvent extends Adapter {
    private BannerCustomEventLoader bannerLoader;
    private InterstialCustomEvent interstitialLoader;
    private NativeCustomEvent nativeLoader;
    private RewardCustomEvent rewardedLoader;
    


    @NonNull
    @Override
    public VersionInfo getSDKVersionInfo() {
        return new VersionInfo(0, 0, 0);
    }


    @NonNull
    @Override
    public VersionInfo getVersionInfo() {
        String versionString = String.valueOf(new VersionInfo(1, 2, 3));
        String[] splits = versionString.split("\\.");

        if (splits.length >= 4) {
            int major = Integer.parseInt(splits[0]);
            int minor = Integer.parseInt(splits[1]);
            int micro = Integer.parseInt(splits[2]) * 100 + Integer.parseInt(splits[3]);
            return new VersionInfo(major, minor, micro);
        }

        return new VersionInfo(0, 0, 0);
    }

    @Override
    public void initialize(Context context,
                           InitializationCompleteCallback initializationCompleteCallback,
                           List<MediationConfiguration> mediationConfigurations) {
        // This is where you will initialize the SDK that this custom
        // event is built for. Upon finishing the SDK initialization,
        // call the completion handler with success.
        initializationCompleteCallback.onInitializationSucceeded();
    }

    @Override
    public void loadBannerAd(
            @NonNull MediationBannerAdConfiguration adConfiguration,
            @NonNull MediationAdLoadCallback<MediationBannerAd, MediationBannerAdCallback> callback) {
        bannerLoader = new BannerCustomEventLoader(adConfiguration, callback);
        bannerLoader.loadAd();
    }
    @Override
    public void loadInterstitialAd(
            @NonNull MediationInterstitialAdConfiguration adConfiguration,
            @NonNull
            MediationAdLoadCallback<MediationInterstitialAd, MediationInterstitialAdCallback>
                    callback) {
        interstitialLoader = new InterstialCustomEvent(adConfiguration, callback);
        interstitialLoader.loadAd();
    }
    @Override
    public void loadNativeAd(
            @NonNull MediationNativeAdConfiguration adConfiguration,
            @NonNull MediationAdLoadCallback<UnifiedNativeAdMapper, MediationNativeAdCallback> callback) {
        nativeLoader = new NativeCustomEvent(adConfiguration, callback);
        nativeLoader.loadAd();
    }

    @Override
    public void loadRewardedAd(
            @NonNull MediationRewardedAdConfiguration mediationRewardedAdConfiguration,
            @NonNull
            MediationAdLoadCallback<MediationRewardedAd, MediationRewardedAdCallback>
                    mediationAdLoadCallback) {
        rewardedLoader =
                new RewardCustomEvent(
                        mediationRewardedAdConfiguration, mediationAdLoadCallback);
        rewardedLoader.loadAd();
    }
}

File: GAM to Admob/kotlin/adapter/AdNetworkCustomEvent.kt


import android.content.Context
import com.example.netlinkiron.kotlin.admanager.BannerCustomEventLoader
import com.example.netlinkiron.kotlin.admanager.InterstialCustomEvent
import com.example.netlinkiron.kotlin.admanager.NativeCustomEvent
import com.example.netlinkiron.kotlin.admanager.RewardCustomEvent
import com.google.android.gms.ads.mediation.*

class AdNetworkCustomEvent : Adapter() {
    private var bannerLoader: BannerCustomEventLoader? = null
    private var interstitialLoader: InterstialCustomEvent? = null
    private var nativeLoader: NativeCustomEvent? = null
    private var rewardedLoader: RewardCustomEvent? = null
    override fun getSDKVersionInfo(): VersionInfo {
        return VersionInfo(0, 0, 0)
    }

    override fun getVersionInfo(): VersionInfo {
        val versionString = VersionInfo(1, 2, 3).toString()
        val splits =
            versionString.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        if (splits.size >= 4) {
            val major = splits[0].toInt()
            val minor = splits[1].toInt()
            val micro = splits[2].toInt() * 100 + splits[3].toInt()
            return VersionInfo(major, minor, micro)
        }
        return VersionInfo(0, 0, 0)
    }

    override fun initialize(
        context: Context,
        initializationCompleteCallback: InitializationCompleteCallback,
        mediationConfigurations: List<MediationConfiguration>
    ) {
        // This is where you will initialize the SDK that this custom
        // event is built for. Upon finishing the SDK initialization,
        // call the completion handler with success.
        initializationCompleteCallback.onInitializationSucceeded()
    }

    override fun loadBannerAd(
        adConfiguration: MediationBannerAdConfiguration,
        callback: MediationAdLoadCallback<MediationBannerAd, MediationBannerAdCallback>
    ) {
        bannerLoader = BannerCustomEventLoader(adConfiguration, callback)
        bannerLoader!!.loadAd()
    }

    override fun loadInterstitialAd(
        adConfiguration: MediationInterstitialAdConfiguration,
        callback: MediationAdLoadCallback<MediationInterstitialAd, MediationInterstitialAdCallback>
    ) {
        interstitialLoader = InterstialCustomEvent(adConfiguration, callback)
        interstitialLoader!!.loadAd()
    }

    override fun loadNativeAd(
        adConfiguration: MediationNativeAdConfiguration,
        callback: MediationAdLoadCallback<UnifiedNativeAdMapper, MediationNativeAdCallback>
    ) {
        nativeLoader = NativeCustomEvent(adConfiguration, callback)
        nativeLoader!!.loadAd()
    }

    override fun loadRewardedAd(
        mediationRewardedAdConfiguration: MediationRewardedAdConfiguration,
        mediationAdLoadCallback: MediationAdLoadCallback<MediationRewardedAd, MediationRewardedAdCallback>
    ) {
        rewardedLoader = RewardCustomEvent(
            mediationRewardedAdConfiguration, mediationAdLoadCallback
        )
        rewardedLoader!!.loadAd()
    }
}